packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -199,37 +217,29 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
         IDisputeGameFactory _disputeGameFactory,
         ISystemConfig _systemConfig,
         ISuperchainConfig _superchainConfig,
-        GameType _initialRespectedGameType
+        IAnchorStateRegistry _anchorStateRegistry
     )
         external
-        initializer
+        reinitializer(2)
     {
         disputeGameFactory = _disputeGameFactory;
         systemConfig = _systemConfig;
         superchainConfig = _superchainConfig;
+        anchorStateRegistry = _anchorStateRegistry;
```

Comment:
We should either change the interface to read the DGF from the ASR to reduce the number of inputs, or have an assertion that `_anchorStateRegistry.disputeGameFactory() == _disputeGameFactory`. Ideally we could just take the SystemConfig as the only input (which we already have) and fetch the rest from there, since that's how the SystemConfig `Addresses` struct was intended to work

Reply:
Resolved by getting rid of the `disputeGameFactory` input and having it pull from the ASR